mapboxgl/utils.py

```diff
@@ -147,3 +153,20 @@ def create_color_stops(breaks, colors='RdYlGn', color_ramps=color_ramps):
                 stops.append([b, ramp[i]])
 
     return stops
+
+
+def img_encode(arr, **kwargs):
+    """Encode ndarray to base64 string image data
+
+    Parameters
+    ----------
+    arr: ndarray (rows, cols, depth)
+    kwargs: passed directly to matplotlib.image.imsave
+    """
+    sio = BytesIO()
+    imsave(sio, arr, **kwargs)
+    sio.seek(0)
+    img_format = kwargs['format'] if kwargs.get('format') else 'png'
+    img_str = base64.b64encode(sio.getvalue()).decode()
+
+    return 'data:image/{};base64,{}'.format(img_format, img_str)
```

Comment:
@vincentsarago it may be good to include an optional `filename` output parameter here.  This would encourage users to output a local file with the encoded image data so that the `mapboxgl` viz class can load the image data asynchronously (versus stick it into a large HTML blob).

Reply:
In fact I think because mapboxgl can read local files we can totally remove this utility function and show the user it can use `imsave`